app/services/reception_service.rb

```diff
@@ -0,0 +1,28 @@
+class ReceptionService
+  def initialize(patient)
+    @patient = patient
+  end
+
+  def check_in
+    current_appointment.update!(check_in: Time.zone.now)
+  end
+
+  def check_out
+    current_appointment.update!(check_out: Time.zone.now)
+
+    create_second_dose_appointment unless current_appointment.second_dose?
+  end
+
+  private
+
+  def current_appointment
+    @patient.last_appointment
+  end
+
+  def create_second_dose_appointment
+    next_appointment = Appointment.find_by(start: current_appointment.start + ENV['SECOND_DOSE_INTERVAL'].to_i.weeks)
```

Comment:
Acho que vale alterarmos pra:

```suggestion
    next_appointment = Appointment.find_by(start: current_appointment.start + ENV['SECOND_DOSE_INTERVAL']&.to_i&.weeks)
```

Porque `nil.to_i` é 0, o que pode causar bugs silenciosos se a variável de ambiente não estiver setada. Já com `nil`, vai dar erro.